### ISSUE TITLE
Is decoding a specific part of a gargantuan file supported efficiently?

### ISSUE THREAD

**Issue by JohnWeisz** [audiocogs/aurora.js#212]
Good afternoon, and thanks for your work on this project.

We are interested in trying out Aurora.js for a new streaming-audio-resource system that requires on-demand random access for specific time ranges. For example, given a 10-hour long mp3 file, I'm wondering if it's possible to extract audio samples from 3:10:00 to 3:10:20, without reading the entire file into memory, or even the part leading up to the requested time-range.

The docs specify 2 use-cases:

 - Running playback at live listening rate to get audio samples (Asset.start and Asset.stop)
 - Decoding the entire thing into a buffer (Asset.decodeToBuffer, this is obviously unusable)

Basically, what I'm wondering about is whether a combination of these 2 is supported.
Thanks.

**Comment by JohnWeisz** [comment]
After some investigation, it seems this should be possible by accessing the _Decoder_ directly, I'll report back with further progress.

**Comment by JohnWeisz** [comment]
From the code it seems this isn't possible in the end (by just inspecting FileSource, it apparently doesn't have anything related to seeking).

But I might be totally wrong here, because to me it doesn't sound viable to not have actual stream seeking and have to download hours of audio before being able to seek to a specific position.

**Comment by jayarjo** [comment]
@JohnWeisz how did it went for you in the end? @devongovett I wonder the same thing.
